File: Systems-for-Design-and-Implementation/Labs/Lab8/ProiectCSharp/TransportNetworking/transport.network/jsonprotocol/Response.cs
using System;
using System.Collections.Generic;
using Transport.Model.transport.model;
using TransportModel.transport.model;
using TransportNetworking.transport.network.jsonprotocol;

namespace TransportNetworking.transport.network.jsonprotocol
{
    [Serializable]
    public class Response
    {
        public ResponseType Type { get; set; }
        public string ErrorMessage { get; set; }
        public ReservationManager ReservationManager { get; set; }
        public Trip Trip { get; set; }
        public Seat Seat { get; set; }
        public Reservation Reservation { get; set; }
        public Trip[] Trips { get; set; }
        public List<Reservation> Reservations { get; set; }
        public List<Seat> Seats { get; set; }

        public override string ToString()
        {
            return $"Response{{type={Type}, reservationManager={ReservationManager}, trip={Trip}, reservation={Reservation}, seat={Seat}}}";
        }
    }
}

File: Systems-for-Design-and-Implementation/Labs/Lab12/ProiectJava/TransportNetworking/src/main/java/transport/network/jsonprotocol/Response.java
package transport.network.jsonprotocol;

import transport.model.Reservation;
import transport.model.ReservationManager;
import transport.model.Seat;
import transport.model.Trip;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

public class Response implements Serializable {
    private ResponseType type;
    private String errorMessage;
    private ReservationManager reservationManager;
    private Trip trip;
    private Seat seat;
    private Reservation reservation;
    private Trip[] trips;
    private List<Reservation> reservations;
    private List<Seat> seats;

    public Response() {
    }

    public ResponseType getType() {
        return type;
    }

    public void setType(ResponseType type) {
        this.type = type;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public ReservationManager getReservationManager() {
        return reservationManager;
    }

    public void setReservationManager(ReservationManager reservationManager) {
        this.reservationManager = reservationManager;
    }

    public Trip getTrip() {
        if (trip != null)
            trip.getDateDeparture();
        return trip;
    }

    public void setTrip(Trip trip) {
        this.trip = trip;
    }

    public Seat getSeat() {
        return seat;
    }

    public void setSeat(Seat seat) {
        this.seat = seat;
    }

    public Reservation getReservation() {
        return reservation;
    }

    public void setReservation(Reservation reservation) {
        this.reservation = reservation;
    }

    public Trip[] getTrips() {
        return trips;
    }

    public void setTrips(Trip[] trips) {
        this.trips = trips;
    }

    public List<Seat> getSeats() {
        return seats;
    }

    public void setSeats(List<Seat> seats) {
        this.seats = seats;
    }

    public List<Reservation> getReservations() {
        return reservations;
    }

    public void setReservations(List<Reservation> reservations) {
        this.reservations = reservations;
    }

    @Override
    public String toString() {
        return "Response{" +
                "type=" + type +
                ", reservationManager=" + reservationManager +
                ", trip=" + trip +
                ", reservation=" + reservation +
                ", seat=" + seat +
                '}';
    }
}